crochet-lang/crochet
# Initial implementation of hover to show inferred type in LSP

## Summary:
TODO:
- [x] add `inferred_type` field to top-level decls (it turns out I just need to specify the `inferred_type` field when inferring patterns)
- [x] get the current version of the code being analyzed which hasn't been saved yet

<img width="638" alt="Capture d’écran, le 2022-12-31 à 23 11 14" src="https://user-images.githubusercontent.com/1044413/210161472-c3b137b9-382c-42d5-80c4-99ce6a584901.png">

I'll write some unit tests in a separate PR since this will likely require some refactoring and I'd like to get the current changes landed.

## Test Plan:
- press F5 to run the VSCode extension in a new instance of VSCode
- open a few different .crochet files
- hover over different things in each file, see the types displayed
- making some edits without saving
- hover over the new code, see the types displayed
